### PR TITLE
fix(ui): modify provider form schema for updating

### PR DESCRIPTION
**Pull request by wwayne**
### Screen record
https://jam.dev/c/c3dd2623-8ddc-45da-8e49-3d14eb418ba2

### What to fix
* Allow the user to update only the display name
* After updating the display name, the top displaying name should be updated as well
* The form should trim the display name

### CODE REVIEW

**Comment by wwayne** [comment]
Need @liangfung help checking if the fix is proper

**Comment by liangfung** [comment]
> Need @liangfung help checking if the fix is proper

Need to test the interaction when the accessToken is not provided in the new creation scenario. I think we can use different schemas for create and update.

**Comment by wwayne** [comment]
The backend API should be good now, just tested
But the frontend still needs to be polished a bit. @liangfung will help to take a look then

**Comment by wwayne** [comment]
Fixed other small issues and updated the PR description
screen record: https://jam.dev/c/c3dd2623-8ddc-45da-8e49-3d14eb418ba2

File: ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/update-github-provider-form.tsx [hunk 1]
```diff
@@ -60,12 +62,19 @@ export const UpdateProviderForm: React.FC<UpdateProviderFormProps> = ({
...
+    const res = await updateGithubRepositoryProvider({
       input: {
         id,
         ...values
       }
     })
+    if (res?.data?.updateGithubRepositoryProvider) {
+      onUpdate?.()
+    } else {
+      toast.error(
```
Comment (liangfung) [inline ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/update-github-provider-form.tsx:74]: No need for manual toasting; if an error occurs, <FormMessage /> will render and display the error message

File: ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/update-gitlab-provider-form.tsx [hunk 2]
```diff
@@ -59,13 +61,20 @@ export const UpdateProviderForm: React.FC<UpdateProviderFormProps> = ({
...
+    const res = await updateGitlabRepositoryProvider({
       input: {
         id,
         ...values
       }
     })
+    if (res?.data?.updateGitlabRepositoryProvider) {
+      onUpdate?.()
+    } else {
+      toast.error(
```
Comment (liangfung) [inline ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/update-gitlab-provider-form.tsx:74]: ditto

Comment (wwayne) [reply ee/tabby-ui/app/(dashboard)/settings/(integrations)/repository/[kind]/detail/components/update-gitlab-provider-form.tsx]: removed `else block`

**Pull request merged**